pytest_html/plugin.py

```diff
@@ -438,21 +438,37 @@ def generate_summary_item(self):
     def _generate_environment(self, config):
         if not hasattr(config, '_metadata') or config._metadata is None:
             return []
+        else:
+            metadata = config._metadata
+            return [html.h2('Environment'),
+                    html.table(
+                        self._create_table_rows_from_dictionary(metadata),
+                        id='environment')]
+
+    def _generate_custom_summary(self, config):
+        custom_summary = {}
+        config.hook.pytest_html_results_summary(custom_summary=custom_summary)
+
+        if not custom_summary:
+            return []
+        else:
+            return [html.table(
+                self._create_table_rows_from_dictionary(custom_summary),
+                id='custom-summary')]
```

Comment:
This limits any summary to being a dictionary that will be formatted as a table. I think this is unecessarily restrictive.

Reply:
Good point, will change anyway when implementing the `pytest_html_results_summary` hook you suggested.